src/ui/public/agg_types/metrics/top_hit.js

```diff
@@ -40,13 +55,9 @@ export const topHitMetricAgg = new MetricAggType({
   params: [
     {
       name: 'field',
+      type: 'field',
       onlyAggregatable: false,
-      filterFieldTypes: function (vis, value) {
-        if (vis.type.name === 'table' || vis.type.name === 'metric') {
-          return true;
-        }
-        return value === 'number';
-      },
+      filterFieldTypes: '*',
```

Comment:
top_hits can work with any field type

Reply:
i think we do, will double check

so, nothing change functionally,

the advantage is that we seperated the filtering into two levels:
- aggconfig will filter the fields down to what it can use (top hits can work with every field)
- we have second layer of filtering in the editor, where we know we can't work with string fields inside a date histogram for example.

we need this seperation for the pipeline, since at the time of data fetching we will have no information about visualization that is going to use this data.